traitlets/config/manager.py

```diff
@@ -51,18 +53,33 @@ def ensure_config_dir_exists(self):
     def file_name(self, section_name):
         return os.path.join(self.config_dir, section_name+'.json')
 
+    def directory(self, section_name):
+        return os.path.join(self.config_dir, section_name+'.d')
+
     def get(self, section_name):
         """Retrieve the config data for the specified section.
 
         Returns the data as a dictionary, or an empty dictionary if the file
         doesn't exist.
         """
         filename = self.file_name(section_name)
+        data = {}
+        if self.read_directory:
+            directory = self.directory(section_name)
+            if os.path.isdir(directory):
+                paths = os.listdir(directory)
+                for path in sorted(paths):
+                    name, ext = os.path.splitext(path)
+                    abspath = os.path.join(directory, path)
+                    if ext == ".json" and name not in data:
```

Comment:
Why the `name not in data` here? Shouldn't all files be loaded?

Reply:
No need for that, leftover from refactoring the code. The idea is that the .json file overrride any values that are set in the .d directory files, but instead I just swapped the order (first read the directory, then read the .json).

Reply:
Perhaps could do a discovery phase and then a load phase, and let bad configs not cause the whole stack to fail, a la:

```py
from glob import glob
...

    def get(self, section_name):
        """Retrieve the config data for the specified section.
        Returns the data as a dictionary, or an empty dictionary if the file
        doesn't exist.
        """
        paths = [self.file_name(section_name)]

        if self.read_directory:
            pattern = os.path.join(self.directory(section_name), '*.json')
            paths = sorted(glob(pattern)) + paths

        data = {}
        for path in paths:
            try:
                if os.path.isfile(path):
                    with io.open(path, encoding='utf-8') as f:
                            recursive_update(data, json.load(f))
            except Exception as err:
                self.log.error(
                    'Couldn\'t update %s configuration from %s: %s',
                    section_name, path, err)

        return data
```

Reply:
Yup, good idea. I think this is an artifact from a misunderstanding, i thought the 'dict' foo.d/bar.json would to into data['bar']. I'll update it.
Not sure if it should accept errors, since it would mean a bad config.